lib/huginn_acumen_product_agent/concerns/inv_product_query_concern.rb

```diff
@@ -0,0 +1,108 @@
+# frozen_string_literal: true
+
+# This module is responsible for reading/processing data recrods from the Inv_Product
+# table in Acumen. This table contains the baseline product information -- name, sku,
+# format, etc.
+module InvProductQueryConcern
+  extend AcumenQueryConcern
+
+  # Fetch/Process the Acumen data,
+  def fetch_inv_product_data(acumen_client, product_ids, digital_format_list)
+    product_data = acumen_client.get_products(product_ids)
+
+    return process_inv_product_response(product_data, digital_format_list)
+  end
+
+  # This function returns an array of Acumen products mapped to Schema.org/Product
+  # objects. We've added additional fields of:
+  #
+  #     *  `isAvailableForPurchase` -- used to control product deletion in external systems
+  #     *  `acumenAttributes` -- Additional acumen data that doesn't have a direct 1:1 field
+  #        on the Product, but may be useful in other platforms
+  def process_inv_product_response(raw_data, digital_format_list)
+    raw_data.map do |p|
+
+      begin
+        product = response_mapper(p, {
+            'Inv_Product.ID' => 'identifier',
+            'Inv_Product.ProdCode' => 'sku',
+            'Inv_Product.Full_Title' => 'name',
+            'Inv_Product.SubTitle' => 'disambiguatingDescription',
+            'Inv_Product.ISBN_UPC' => 'isbn',
+            'Inv_Product.Pub_Date' => 'datePublished',
+            'Inv_Product.Next_Release' => 'releaseDate',
+        })
+
+        product['@type'] = 'Product'
+        product['isTaxable'] = get_field_value(p, 'Inv_Product.Taxable') == '1'
+        product['isAvailableForPurchase'] = get_field_value(p, 'Inv_Product.Not_On_Website') == '0'
+        product['acumenAttributes'] = {
+          'info_alpha_1' => get_field_value(p, 'Inv_Product.Info_Alpha_1'),
+          'info_boolean_1' => get_field_value(p, 'Inv_Product.Info_Boolean_1'), # is_available_on_formed
+        }
+        product['additionalProperty'] = [
+          {
+              '@type' => 'PropertyValue',
+              'propertyID' => 'is_master',
+              'value' => get_field_value(p, 'Inv_Product.OnWeb_LinkOnly') == '0',
```

Comment:
@mrogers-5s We need this to be = to "1" if true and "0" if false and I do not think we want to do that in its own agent. Can we have this fixed here?

Reply:
No, the above logic is correct as is.

The Acumen field is `true` (`1`) when the product should be hidden on the site, and `false` (`0`) when it is a master product. So, `is_master` should be true when the Acumen field is `0`.